app/Services/Geocoder/PositionstackGeocoder.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services\Geocoder;
+
+use Illuminate\Support\Facades\Http;
+
+class PositionstackGeocoder implements GeocoderInterface {
+    public function geocode(string $address): ?array {
+        $response = Http::get('http://api.positionstack.com/v1/forward', [
```

Comment:
I suggestion not using Facade. This makes it not unit testable. You could better Dependancy Inject an HttpClient and use that. This way , when you are unit testing, you can inject a mock client and assert the API is called correctly and the Geocoder responds correcty to certain responses.

Reply:
This `Http` facade allows unit testing: https://laravel.com/docs/8.x/http-client#testing

Reply:
Hmmm, Using Dependancy Injection will make it more implementation free, and testable without Laravel itself.

Reply:
@blackshadev I've been reading up a bit on Facades (I actually understand how they work now, back when I started AAS this was still deep magic to me) and on people's concerns with them. But I'm still not convinced to switch to DI in this case.

We've already covered testing, so (I think) the only concern left is the tight coupling to Laravel. But we're using a framework to help us right? I can't really envision a scenario in which we would rewrite all of AAS in a different framework, and even if we would, it'll be a lot of work regardless of this. I think the benefits of the framework outweigh the architectural concern here.

Besides, to make it truly implementation free, you'd need an interface to which multiple http clients adhere, and I don't think one exists. So in the case of switching http clients (which also is probably not going to happen) we'd still need to rewrite code.

Reply:
Making something less coupled the an external package (like Laravel) doesn't just mean that you can migrate away from it more easily. But more importantly, the cost of maintainability drastically decreases. Because if you update dependencies (for instance upgrading Laravel). You will not have to worry that it breaks. 
Regarding the interface, there is a "generic" http interface in de PSR spec I believe.

But I can live with using the facade. Functionally it works, and it is testable. I do suggest having a look at how DI works (in laravel), because it will makes things better, and it is quite universal as a programming paradigma nowadays.

Reply:
I already used DI for injecting the geocoder into the event listener that handles member updates :)  so I got that down.